rosbag2_transport/test/rosbag2_transport/test_play_timing.cpp

```diff
@@ -191,3 +191,64 @@ TEST_F(PlayerTestFixture, playing_respects_rate)
     ASSERT_THAT(replay_time, Gt(message_time_difference));
   }
 }
+
+TEST_F(PlayerTestFixture, playing_respects_delay)
+{
+  auto primitive_message1 = get_messages_strings()[0];
+  primitive_message1->string_value = "Hello World 1";
+
+  auto primitive_message2 = get_messages_strings()[0];
+  primitive_message2->string_value = "Hello World 2";
+
+  auto message_time_difference = std::chrono::seconds(1);
+  auto topics_and_types =
+    std::vector<rosbag2_storage::TopicMetadata>{{"topic1", "test_msgs/Strings", "", ""}};
+  std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>> messages =
+  {serialize_test_message("topic1", 0, primitive_message1),
+    serialize_test_message("topic1", 0, primitive_message2)};
+
+  messages[0]->time_stamp = 100;
+  messages[1]->time_stamp =
+    messages[0]->time_stamp + std::chrono::nanoseconds(message_time_difference).count();
+
+  float delay_margin = 1.0;
+
+  // Sleep 5.0 seconds after play
+  {
+    play_options_.delay = 5.0;
+    std::chrono::duration<float> delay(play_options_.delay);
+    std::chrono::duration<float> delay_uppper(play_options_.delay + delay_margin);
+
+    auto prepared_mock_reader = std::make_unique<MockSequentialReader>();
+    prepared_mock_reader->prepare(messages, topics_and_types);
+    auto reader = std::make_unique<rosbag2_cpp::Reader>(std::move(prepared_mock_reader));
+    auto player = std::make_shared<rosbag2_transport::Player>(
+      std::move(
+        reader), storage_options_, play_options_);
+    auto start = std::chrono::steady_clock::now();
+    player->play();
+    auto replay_time = std::chrono::steady_clock::now() - start;
+
+    ASSERT_THAT(replay_time, Gt(message_time_difference + delay));
+    ASSERT_THAT(replay_time, Lt(message_time_difference + delay_uppper));
+  }
+
+  // Invalid value should result in playing at default delay 0.0
+  {
+    play_options_.delay = -5.0;
+    std::chrono::duration<float> delay_uppper(delay_margin);
+
+    auto prepared_mock_reader = std::make_unique<MockSequentialReader>();
+    prepared_mock_reader->prepare(messages, topics_and_types);
+    auto reader = std::make_unique<rosbag2_cpp::Reader>(std::move(prepared_mock_reader));
+    auto player = std::make_shared<rosbag2_transport::Player>(
+      std::move(
+        reader), storage_options_, play_options_);
```

Comment:
Nitpick. It would be better if 
```cpp
    std::move(
        reader), storage_options_, play_options_);
```
would be on one line
```cpp
    std::move(reader), storage_options_, play_options_);
```

Reply:
OK, I modified all same part in https://github.com/ros2/rosbag2/pull/789/commits/75eca9f527e9a5fcba9d88af42420715d8f13a07.